### PR TITLE
MAINT: Use a dictionary for function name synonyms

### PR DESCRIPTION
**Pull request by endolith**
Gets rid of duplication of information and long elif chain, minimizing future problems like https://github.com/e-q/scipy/commit/55c2ca84381214759c26589013e38cc95a0b02ec

(also some minor PEP8 fixes)

### CODE REVIEW

**Comment by rgommers** [comment]
Good idea. TravisCI tests failed because the pep8 checked frowns on ``multiple spaces after ':'``

**Comment by rgommers** [comment]
(can you squash the commits and start the commit message with ``MAINT``?)

File: scipy/signal/windows.py [hunk 1]
```diff
@@ -1569,6 +1570,42 @@ def exponential(M, center=None, tau=1., sym=True):
...
+    ('hamming', 'hamm', 'ham'):                     (hamming, False),
+    ('hanning', 'hann', 'han'):                     (hann, False),
+    ('kaiser', 'ksr'):                              (kaiser, True),
+    ('nuttall', 'nutl', 'nut'):                     (nuttall, False),
+    ('parzen', 'parz', 'par'):                      (parzen, False),
+    ('slepian', 'slep', 'optimal', 'dpss', 'dss'):  (slepian, True),
+    ('triangle', 'triang', 'tri'):                  (triang, False),
+    ('tukey', 'tuk'):                               (tukey, True),
+}
+
```
Comment (rgommers) [inline scipy/signal/windows.py]: maybe add a comment here, something like ``# filling dict with all valid window name strings``

Comment (rgommers) [inline scipy/signal/windows.py]: and similar for the ``needs_param`` set.

File: scipy/signal/windows.py [hunk 2]
```diff
@@ -1569,6 +1570,42 @@ def exponential(M, center=None, tau=1., sym=True):
     return w
 
 
+win_equiv_raw = {
```
Comment (rgommers) [inline scipy/signal/windows.py]: Would be good to prepend underscores to ``win_equiv_raw``, ``win_equiv``. ``needs_param``

**Comment by endolith** [comment]
ok, did those.  should I remove the extra spaces?

**Comment by rgommers** [comment]
@endolith yes please. you can check locally that it's fixed with ``$ pep8 scipy`` from the root of your repo.

**Comment by endolith** [comment]
ok changed

**Comment by rgommers** [comment]
Looks good now, merged. Thanks @endolith

**Pull request merged**